core/trino-spi/src/main/java/io/trino/spi/connector/JoinCondition.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.expression.ConnectorExpression;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class JoinCondition
+{
+    public enum Operator
+    {
+        EQUAL("="),
+        NOT_EQUAL("<>"),
+        LESS_THAN("<"),
+        LESS_THAN_OR_EQUAL("<="),
+        GREATER_THAN(">"),
+        GREATER_THAN_OR_EQUAL(">="),
+        IS_DISTINCT_FROM("IS DISTINCT FROM");
+
+        private final String value;
+
+        Operator(String value)
+        {
+            this.value = value;
+        }
+
+        public String getValue()
+        {
+            return value;
+        }
+    }
+
+    private final Operator operator;
+    private final ConnectorExpression leftExpression;
+    private final ConnectorExpression rightExpression;
```

Comment:
What's this?

Reply:
A Variable, modelled as in agg pushdown. 
When connector expressions are expanded so support eg arithmetics, but we don't roll out newer join spi yet, this can give us an option to push more here.

Reply:
I would just make it a Variable. Supporting arithmetics means supporting arbitrary expressions, at which point we'll just switch to the approach of pushing down the entire join condition and get rid of this JoinCondition class.

Reply:
I don't plan to expose arithemtics (they are not expressible in connector expression anyway), but when we add them, this API can still be used to take advantage of them _while_ we work on the new API.
I don't think there is any harm here, any downside to having ConnectorExpression here, is there?

Reply:
>  any downside to having ConnectorExpression here, is there?

Just a slight lack of clarity, and something that's not clear will actually be needed. We can always extend it later to support more generic expressions if necessary (e.g., change the type of the fields, add new getters and deprecate old ones).

Reply:
That would be a breaking change for this API. 

If we want to go _really_ simple, we can put `ConnectorColumnHandle` here and remove all the assignments staff.
Works for my short term plans, but i don't see why we would out ourselves into a corner. Clarity concern can be resolved with a javadoc.